### ISSUE TITLE
conflicting/ambiguous licensing information (LGPL)

### ISSUE THREAD

**Issue by onlyjob** [residualvm/residualvm#1146]
According to `COPYING.LGPL` the following files are licensed under LGPL-2.1 ("only" or including "any later version" ?):

 * math/quat.cpp
 * math/matrix4.cpp

However the above files contain GPL-2+ header. Perhaps those files' headers should be corrected to retain LGPL-2.1 terms?
Please clarify.

**Comment by aquadran** [comment]
that files are under GPL-2+ license, some portions of code is based on code using LGPL-2.1 license.
I didn't add code myself, so I have to look into history of that files.

**Comment by onlyjob** [comment]
A file can not be available under GPL-2+ and LGPL-2.1 at the same time. Which license covers those files? I don't remember if LGPL-2.1 allows re-licensing under stricter terms of GPL-2+ without explicit permission of the copyright owners therefore I suspect that the license of files in question is just LGPL-2.1, possibly LGPL-2.1+.

**Comment by aquadran** [comment]
Based on history of this file, content of two functions are based on LGPL content. Later we relicensed our LGPL code to GPL, so code based on LGPL (not our origin) was left under original license.
I'm not sure now, but probably you are right, that files can not be under the same headers.
If it's indeed not correct we will probably move that code to separated file.

**Comment by aquadran** [comment]
regarding matrix4.cpp it's similar situation

**Comment by somaen** [comment]
Is there specifically a law/rule that says that a license governs a whole file? As far as I know, including LGPL2-code is allowed under GPL2, without the GPL2-license taking precedence, however it is also allowable to convert anything that is LGPL2.1 (not sure right now if LGPL2 is also covered) to GPL2 without asking permission (as the LGPL2.1 license clearly states that this is allowed, this is incidentally how we changed the grim-engine from LGPL to GPL originall).

We have a norm of retaining reference when we use larger pieces of code in this fashion, and clearly indicating what the original license was (and thus what rights we had to use said code) is just common sense then.

Also, a file can be available under as many license that whoever created it wants at any time, the only problem is specifying under which terms it is under what license (and avoiding that such terms conflict with the license itself), see for instance qt which is available both under GPL, and non-GPL.

So, to summarize, relicensing is fine (atleast if it is LGPL2.1), as I remember clearly from the grim-engine that such a clause does indeed exist. Also, we might be clearer as to what terms it follows in specific.

Also, our resulting binary ends up being under the GPL2+, while the parts it consists of are indeed a variety of (GPL-compatible) licenses.

**Comment by onlyjob** [comment]
Right, LGPL-2.1 says that releasing as GPL-2+ is OK:

>   You may opt to apply the terms of the ordinary GNU General Public
License instead of this License to a given copy of the Library.  To do
this, you must alter all the notices that refer to this License, so
that they refer to the ordinary GNU General Public License, version 2,
instead of to this License.  (If a newer version than version 2 of the
ordinary GNU General Public License has appeared, then you can specify
that version instead if you wish.)  Do not make any other change in
these notices.

Hence those files are correctly fall into the scope of GPL-2+ as part of the project.

I reckon to avoid confusion it would be enough to remove `COPYING.LGPL` file (which incorrectly says that files are under LGPL-2.1).

**Comment by somaen** [comment]
Not entirely, there is LGPL code in engines/grim/update/mspack.cpp (although we've failed to mention that in COPYING.LGPL). That is LGPL2.1 only, and thus can be converted to GPL2 (only) easily, this does however have the side-effect of making the resulting binary GPL2 (only), instead of GPL2+ _IF_ the grim-engine is enabled during compilation.

**Comment by onlyjob** [comment]
OK, then perhaps just removing files "math/quat.cpp" and "math/matrix4.cpp" from `COPYING.LGPL` should be sufficient...

@somaen, You probably meant `engines/grim/update/mscab.cpp`? According to its header it is GPL-2+... Same problem (inconsistent licensing info in header)?

**Comment by YakBizzarro** [comment]
About mscab.cpp, if I remember correctly, only the function MsCabinet::Decompressor::checksum comes from libmspack. The others parts are heavily inspired from KCabinet, but they don't borrow any code.

**Comment by aquadran** [comment]
situation with mscab.cpp resolved, all libmspack code was removed

**Comment by aquadran** [comment]
some parts of math/quat.cpp and matrix4.cpp code converted to GPL, this resolve header inconsistency

**Comment by aquadran** [comment]
we will release 0.2.1 in the end of next week, that changes will also be included
